lua/elixir/projectionist/init.lua

```diff
@@ -32,11 +32,73 @@ local config = {
       },
     },
     ["test/**/controllers/*_controller_test.exs"] = {
+      type = "test",
       alternate = "lib/{dirname}/controllers/{basename}_controller.ex",
+      template = {
+        "defmodule {dirname|camelcase|capitalize}.{basename|camelcase|capitalize}ControllerTest do",
+        "  use {dirname|camelcase|capitalize}.ConnCase, async: true",
+        "end",
+      },
+    },
+    ["lib/**/controllers/*_html.ex"] = {
+      type = "html",
+      alternate = "test/{dirname}/controllers/{basename}_html_test.exs",
+      template = {
+        "defmodule {dirname|camelcase|capitalize}.{basename|camelcase|capitalize}HTML do",
+        "  use {dirname|camelcase|capitalize}, :html",
+        "",
+        "  embed_templates {basename|snake_case}_html/*",
+        "end",
+      },
+    },
+    ["test/**/controllers/*_html_test.exs"] = {
       type = "test",
+      alternate = "lib/{dirname}/controllers/{basename}_html.ex",
       template = {
         "defmodule {dirname|camelcase|capitalize}.{basename|camelcase|capitalize}ControllerTest do",
         "  use {dirname|camelcase|capitalize}.ConnCase, async: true",
+        "",
+        "  alias {dirname|camelcase|capitalize}.{basename|camelcase|capitalize}HTML",
+        "end",
+      },
+    },
+    ["lib/**/components/*_component.ex"] = {
```

Comment:
this i would for sure just name `lib/**/components/*.ex`.

I don't think anyone would be naming their components in a way that would be `<TableComponent.render {@foo}>...</TableComponent.render>`

(well, at least I wouldn't ever name a component like that).

Reply:
Hm, okay so you'd also change the module definition to 
```
        "defmodule {dirname|camelcase|capitalize}.{basename|camelcase|capitalize} do",
        "  use Phoenix.Component",
        "end",
```
or would you scope it differently?

Looking through other open source repos

https://github.com/BeaconCMS/beacon/blob/main/lib/beacon_web/live/admin/media_library_live/upload_form_component.ex
Dockyard seems to use _component on BeaconCMS

https://github.com/adoptoposs/adoptoposs/blob/develop/lib/adoptoposs_web/live/project_component.ex
This one seems to too!

https://github.com/qhwa/bonfire/blob/master/lib/bonfire_web/live/reading_state_componnet.ex
Same here

https://github.com/fremantle-industries/slurpee/blob/main/lib/slurpee_web/components/block_number_component.ex
Also here!

Reply:
We use Surface at my job, which actually has module components, and we don't name them with a "component" suffix 🤷.

But, I'll note that excluding the last link (which was last modified two years ago), those are all examples of LiveComponents, not function components.

I can add a PR that allows to easily override these defaults if that is what you'd like to see.

Reply:
That is true, I suppose what might make sense would be another set of projections that detect `_component.ex` in the live directory and make live component tests?

Reply:
Yeah I did realize that there wasn't one for live components.

nor the new _json convention.

a PR for either/both of those would great!